linux/libio/Rate.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2015 by Robert Haschke <rhaschke at techfak dot uni-bielefeld dot de>
```

Comment:
Since you admitted this work is derived from [ROS rate](http://docs.ros.org/en/noetic/api/rostime/html/rate_8cpp_source.html)
I have a problem with this new license not retaining the original copyright notice as requested by the original license the code was picked (and modified) from. Deciding for a different (but compatible) license of your derived work is permitted by BSD (see https://dwheeler.com/essays/floss-license-slide.html) , but under the condition that the copyright is kept which I don't see here.
also see https://www.patrick-wied.at/blog/how-to-correctly-use-code-you-didnt-write

I would also keep the original author as courtesy, but it is not required
```suggestion
 * Copyright (c) 2009, Willow Garage, Inc.
   Copyright (C) 2015 by Robert Haschke <rhaschke at techfak dot uni-bielefeld dot de>
```